src/v/cloud_storage/anomalies_detector.cc

```diff
@@ -192,4 +192,71 @@ anomalies_detector::result::operator+=(anomalies_detector::result&& other) {
     return *this;
 }
 
+void anomalies_detector::scrub_segment_meta(
```

Comment:
The checks look correct to me, but we're also have this metadata correctness check - https://github.com/redpanda-data/redpanda/blob/dev/src/v/cloud_storage/partition_manifest.cc#L921
It's a bit more basic so maybe it makes sense to improve teh check in the partitoin_manifest and make it consistent with this check?

Reply:
There is also this spot: https://github.com/redpanda-data/redpanda/blob/dev/src/v/archival/ntp_archiver_service.cc#L1736

Reply:
Changed the partition manifest to check new segments with the same logic in a new commit
> cloud_storage: use new scrub logic on segment add

I didn't touch the change in archival since we don't have a full segment meta there. Also, I'm unsure about the purpose of that check. Is it to just avoid replicating a bad segment? The "real check" is done in the partition manifest.